### PR TITLE
Forward Details

### LINKED ISSUE #24: Routing Key editorial todo (fka DID Doc Requirements)
DID and DID Document Requirements
DIDs can by any method that supports the required key types and a service block of the following format:

```
"service": [{
    "id": "#someid",
    "type": "didcommmessaging",
    "serviceEndpoint": "https://example.com/endpoint",
    "routingKeys": ["did:example:somemediator#somekey"]
  }]
```
If ledger reliance is an issue, Peer DIDs (particularly method 2) may be used.

### PR DESCRIPTION
**Pull request by TelegramSam**
Fixes #24

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -86,12 +86,38 @@ Both parties MUST have a `service` block containing the following properties:
...
+
+#### Routing Keys
+
+Routing Keys are used to enable message routing to agents unable to provide a direct service endpoint. Routing is arranged by the message recipient and communicated in the Service Endpoint as detailed above. 
+
+For each DID key reference present in the `routingKeys` list in order, take the encrypted message, wrap it in a forward message as detailed below, and encrypt to the DID key reference. The newly encrypted message becomes the message to transmit to the listed `serviceEndpoint` or encrypted to the next Key in the `routingKeys` list.
+
+Forward message structure:
+
+```jsonc
```
Comment (brentzundel) [inline spec/spec.md:103]:
```suggestion
```json5
```

**Review by brentzundel — APPROVED** [review]
Thank you for the PR!

I think SpecUp understands json5 rather than jsonc
otherwise LGTM

File: spec/spec.md [hunk 2]
```diff
@@ -86,12 +86,38 @@ Both parties MUST have a `service` block containing the following properties:
   }]
 ```
 
-TODO: Explain routing keys. Each property and normative status:
 - service block must be present
 - `id` MUST contain a unique id
 - `type` MUST be `DIDCommMessaging`
 - `serviceEndpoint` MUST be a resolvable URI
-- `routingKeys` MUST contain valid routing keys (ref directly to DIDComm spec section about routing keys)
+- `routingKeys` MUST contain 0 or more valid routing keys. See next section for details.
```
Comment (brianorwhatever) [inline spec/spec.md:93]: if there is 0 routing keys how does the sender encrypt?

Comment (brianorwhatever) [reply spec/spec.md:93]: I think maybe I've been misunderstanding. I was under the impression that all of the encryption in didcomm is done through routing keys. Is the original message encrypted to a different key in the recipients diddoc? If so, which?

**Review by OR13 — APPROVED** [review]

**Comment by TelegramSam** [comment]
cleanup language around 0 or more routing keys

**Pull request merged**